src/main/java/fr/ens/biologie/genomique/eoulsan/modules/diffana/DESeq2Checker.java

```diff
@@ -36,19 +38,46 @@ static boolean checkExperimentDesign(Experiment experiment) throws EoulsanExcept
     static boolean checkExperimentDesign(Experiment experiment, boolean throwsException) throws EoulsanException {
 
         requireNonNull(experiment, "Experiment argument cannot be null");
-
         final ExperimentMetadata emd = experiment.getMetadata();
-
+        // Check the comparison string, if exists it is a complex design
         if (emd.containsComparisons()) {
-
             // Check if the comparison value is correct
             for (String c : emd.getComparisons().split(";")) {
                 String[] splitC = c.split(":");
+                // Check if there is not more than one value per comparison
                 if (splitC.length != 2) {
                     return error("Error in "
                             + experiment.getName()
                             + " experiment, comparison cannot have more than 1 value: " + c, throwsException);
                 }
+
+                // Get the name of all Metadata keys
+                String[] metakeys = DesignUtils.getExperimentSampleAllMetadataKeys(experiment).toArray(new String[0]);
+                // Get each condition in the comparison string
+                String[] listConditionsInComparisonString = splitC[1].split("(%)|(_vs_)");
+                // Get every sample value for each Metadata key, and get every possible condition by merging the strings
+                ArrayList<String> possibleConditions = new ArrayList<String>();
```

Comment:
Code can be simplified using:
`List<String> possibleConditions = new ArrayList<>();`